pkg/gemspec/testdata/json-2.3.0.gemspec

```diff
@@ -0,0 +1,37 @@
+#!/usr/bin/env jruby
+require "rubygems"
+
+spec = Gem::Specification.new do |s|
+  s.name = "$json"
```

Comment:
Could you tell me the spec of the name field? Any documentation? I want to know why `$` is accepted here.

Reply:
you can refer https://guides.rubygems.org/specification-reference/#name. It suggests formal syntax.
To explore, I tried below commented regexes on image kankush/ruby:v2. This image has famous ruby packages installed.
I did not miss the name from any gemspec file. But I still think regexes should be used, as this is not strict format. 

$json is no different than simple package name. So should be considered.